src/test/java/seedu/address/logic/parser/AddressBookParserTest.java

```diff
@@ -106,4 +113,27 @@ public void parseCommand_unknownCommand_throwsParseException() {
         assertThrows(ParseException.class, MESSAGE_UNKNOWN_COMMAND, () ->
             parser.parseCommand("unknownCommand"));
     }
+
+    @Test
+    public void previousCommands_empty() {
+        //Only neccsary as it is currently static (To be fixed)
+        for (int i = 0; i < 20; i++) {
+            parser.getPreviousCommands();
+        }
+        assertEquals("", parser.getPreviousCommands());
+    }
+
+    @Test
+    public void previousCommands_nonEmpty() {
+        AddressBookParser parserWithCommands = new AddressBookParser();
+        parserWithCommands.addCommand("list");
+        assertEquals("list", parserWithCommands.getPreviousCommands());
+    }
+
+    @Test
+    public void addCommand() {
+        AddressBookParser parserAddCommands = new AddressBookParser();
+        parserAddCommands.addCommand("list");
+        assertEquals("list", parserAddCommands.getPreviousCommands());
+    }
```

Comment:
These two test cases look rather similar, perhaps one of the tests is redundant in this case? Perhaps more test cases can be written as well.